components/ai_chat/core/common/mojom/untrusted_frame.mojom

```diff
@@ -56,4 +56,16 @@ interface UntrustedUIHandler {
   OpenURLFromResponse(url.mojom.Url url);
 
   BindParentPage(pending_receiver<ParentUIFrame> parent_frame);
+
+  // Delete a specific memory from user memory storage
+  DeleteMemory(string memory);
+
+  // Get all current memories from storage
+  GetMemories() => (array<string> memories);
+
+  // Check if memory feature is currently enabled
+  GetMemoryEnabled() => (bool enabled);
```

Comment:
If we're going to keep these, then it seems like they should be combined to a single call or, preferably just provided if in a `UntrustedUIState` struct as a response to GetState or just `BindUntrustedUI()` if we need that to provide an event to. Let's try to keep the frame as lightweight as possible though, to free airspace for the conversation entry streaming messages.

Reply:
Will remove these and move back to HasMemory.